crates/runtime/src/embeddings/execution_plan.rs

```diff
@@ -290,7 +292,12 @@ pub(crate) async fn compute_additional_embedding_columns(
 
             Arc::new(vectors) as ArrayRef
         } else {
-            let fixed_size_array = get_vectors(arr_iter, &**model, cfg.vector_size).await?;
+            let fixed_size_array = if model.supports_sync_embeddings() {
+                get_vectors_in_process(arr_iter, &**model, cfg.vector_size)?
```

Comment:
Given that this will potentially block the thread, we may need to run this in a [tokio::task::spawn_blocking](https://docs.rs/tokio/latest/tokio/task/fn.spawn_blocking.html)